### ISSUE TITLE
Issue in the restclient.go

### ISSUE THREAD

**Issue by d0rc** [henomis/qdrant-go#2]
First of all - this client is much better, compared to official one. Thank you, sir!

The issue is in the bad code of `restclientgo` dependency, namely in `restclient.go`:

```golang
func (r *RestClient) do(ctx context.Context, method httpMethod, request Request, response Response) error {

       ...........

	response.SetStatusCode(httpResponse.StatusCode)
	if httpResponse.StatusCode >= 400 {
		response.SetBody(httpResponse.Body)
		return nil
	}
```

In case error code is 400, client returns "ok" and no detailed information about the error server got parsing the request. I had to modify the code in order to read the error message and to fix my client usage - I was sending empty string in `distance` field, and got cryptic error in logs of qdrant itself - error code 400.

The error message I got was:

```json
{"error":"Format error in JSON body: data did not match any variant of untagged enum VectorsConfig at line 1 column 39"},"time":0.0}
```

It took me some time to figure out it was because of empty string instead of distance.

**Comment by henomis** [comment]
Thank you, I'll check this soon.

**Comment by henomis** [comment]
@d0rc could you use `v1.0.3-alpha.1` please? Error should be decoded with 400 status code also. Use `Status` to see the message.

**Comment by henomis** [comment]
Fix has been done in v1.1.0
